### PR TITLE
Update to use Ember.Array API...

### LINKED ISSUE #55: ember-collection doesn't support Ember arrays
I fetch a list of ingredients using this computed property:

    results: Ember.computed('searchQuery', function() {
      return this.store.query('ingredient', { filter: searchQuery, limit: 150 });
    }

Then I can iterate trough the `results` using:

    {{#each results as |result|}}
      {{result.name}}
    {{/each}}

------

Now I'd like to use `{{ember-collection}}` but when I try to use it as follow: (it's basically the example on the readme of the component)

    {{#ember-collection
      items=results
      height=500
      width=800
      cell-layout=(fixed-grid-layout 800 50) as |item index|
    }}
      {{name}}
    {{/ember-collection}}

the ember-collection component displays just an empty list.

What am I doing wrong?

### PR DESCRIPTION
**Pull request by lukemelia**
… so that it will work with ArrayProxy, PromiseArray, etc

Fixes #55

### CODE REVIEW

**Comment by lukemelia** [comment]
@krisselden @mmun @raytiley would love a code review on this

File: addon/components/ember-collection.js [hunk 1]
```diff
@@ -80,17 +81,24 @@ export default Ember.Component.extend({
...
+        });
       }
-
+      this._rawItems = rawItems;
+      var items = Ember.A(rawItems);
       this.set('_items', items);
 
-      if (items && items.addObserver) {
-        items.addObserver('[]', this, this._needsRevalidate);
+      if (items && items.addEnumerableObserver) {
```
Comment (krisselden) [inline addon/components/ember-collection.js]: What is the point of this?

Comment (lukemelia) [inline addon/components/ember-collection.js]: With the previous code, calling addObserver with a property of `'[]'`raised on error when used with an ArrayProxy.

Comment (krisselden) [inline addon/components/ember-collection.js]: That sounds like a bug in ArrayProxy to be honest, also, addEnumerableObserver is soft deprecated and marked as private in 2.0.  We can use addArrayObserver but to be honest I was adding a indirection here called dataSource so people could roll their own sparse array as part of the layout bin refactor but I'd like to pull this in for the mean time.

**Pull request merged**